### PR TITLE
feat(connect):Support (min/max) constraints for DataGen connector

### LINKED ISSUE #2747: Support (min/max) constraints for DataGen connector

### PR DESCRIPTION
**Pull request by D2Lark**
## What's changed and what's your intention?

As title

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Add the 'user-facing changes' label if your PR contains changes that are visibile to users (optional) 

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

Close #2747

### CODE REVIEW

File: src/connector/src/datagen/mod.rs [hunk 1]
```diff
@@ -34,6 +36,8 @@ pub struct DatagenProperties {
         default = "default_rows_per_second"
     )]
     pub rows_per_second: String,
+    #[serde(flatten)]
+    fields: HashMap<String, String>,
```
Comment (TennyZhuang) [inline src/connector/src/datagen/mod.rs:40]: Better to add some comments here, e.g. Options related to fields

File: src/connector/src/datagen/source/field_generator/mod.rs [hunk 2]
```diff
@@ -12,72 +12,93 @@
...
+mod varchar;
+
 use anyhow::Result;
 use rand::{thread_rng, Rng};
 use risingwave_common::types::DataType;
 use serde_json::{json, Value};
+pub use timestamp::*;
+pub use varchar::*;
+
+pub trait FieldGenerator {
```
Comment (TennyZhuang) [inline src/connector/src/datagen/source/field_generator/mod.rs:25]: How about `NumericFieldGenerator`? I guess it cannot be applied to Timestamp/VARCHAR. And better to add some comments for pub trait.

File: src/connector/src/datagen/source/field_generator/mod.rs [hunk 3]
```diff
@@ -12,72 +12,93 @@
...
+    fn with_sequence(min: Option<String>, max: Option<String>) -> Result<Self>
+    where
+        Self: Sized;
+    fn with_random(start: Option<String>, end: Option<String>) -> Result<Self>
+    where
+        Self: Sized;
+    fn generate(&mut self) -> Value;
+}
+
+// Generator of this '#' field. Can be 'sequence' or 'random'.
```
Comment (TennyZhuang) [inline src/connector/src/datagen/source/field_generator/mod.rs:35]:
```suggestion
/// Generator of this '#' field. Can be 'sequence' or 'random'.
```

File: src/connector/src/datagen/source/field_generator/timestamp.rs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+use rand::{thread_rng, Rng};
+use serde_json::{json, Value};
+
+use crate::datagen::source::field_generator::FieldGenerator;
+
+pub struct TimestampField {
+    max_past: Duration,
+    local_now: NaiveDateTime,
+}
+impl TimestampField {
```
Comment (TennyZhuang) [inline src/connector/src/datagen/source/field_generator/timestamp.rs:28]:
```suggestion

impl TimestampField {
```

File: src/connector/src/datagen/source/field_generator/varchar.rs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use rand::distributions::Alphanumeric;
+use rand::Rng;
+use serde_json::{json, Value};
+
+use super::DEFAULT_LENGHT;
```
Comment (shanicky) [inline src/connector/src/datagen/source/field_generator/varchar.rs:20]: typo?

**Review by shanicky — CHANGES_REQUESTED** [review]

File: src/connector/src/datagen/source/field_generator/varchar.rs (hunk 5 above)
Comment (D2Lark) [reply src/connector/src/datagen/source/field_generator/varchar.rs:20]: I will fix that..

**Comment by TennyZhuang** [comment]
rustdoc recommends `///` to write doc comments and `//` to write normal comments.

https://doc.rust-lang.org/rust-by-example/meta/doc.html

**Comment by TennyZhuang** [comment]
Please resolve conflicts

**Review by TennyZhuang — APPROVED** [review]

**Review by shanicky — APPROVED** [review]
LGTM

**Pull request merged**